### PR TITLE
Ashyen and Keleth to SAI

### PR DESCRIPTION
**Pull request by velinath**
**Changes proposed**:

- Another SAI migration.

**Target branch(es)**: 335

**Issues addressed**: Fixes #

**Tests performed**: (Does it build, tested in-game, etc)

**Known issues and TODO list**:

- [ ] player->TalkedToCreature not triggered via SAI

### CODE REVIEW

File: sql/updates/world/3.3.5/2017_09_25_99_world.sql [hunk 1]
```diff
@@ -0,0 +1,40 @@
+-- Add gossip text to menu
+
+DELETE FROM `gossip_menu_option` WHERE `MenuID` IN (7559,7560)
```
Comment (tkrokli) [inline sql/updates/world/3.3.5/2017_09_25_99_world.sql]: Missing semicolon at the end :-p

File: sql/updates/world/3.3.5/2017_09_25_99_world.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
+-- Add gossip text to menu
+
```
Comment (tkrokli) [inline sql/updates/world/3.3.5/2017_09_25_99_world.sql]: Redundant blank line, unless you are reserving the space for a future comment.

**Comment by tkrokli** [comment]
From the SQL: `-- TODO: We also need to set the talkedtocreature flag per the existing c++ script for quest 9785 blessings of the ancients`

Hmm, is this in any way related to https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Kalimdor/zone_the_barrens.cpp#L68
that I once edited & updated for `npc_beaten_corpse` back when I was unable to find a SAI solution?

**Comment by velinath** [comment]
@tkrokli probably? I'd _like_ to make it work with SAI obviously so reaching out to devs on IRC to see if there's a solution but yes it looks similar. All of the rest of the behavior can be accomplished using SAI and conditionals

File: sql/updates/world/3.3.5/2017_09_25_99_world.sql [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+INSERT INTO `gossip_menu_option` (`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES
+(7559,0,0,'Grant me your mark, wise ancient.',14739,1,3,0,0,0,0,'',0,-1),
+(7560,0,0,'Grant me your mark, mighty ancient.',14741,1,3,0,0,0,0,'',0,-1);
+
+-- Migrate NPC to use SmartAI
+UPDATE `creature_template` SET `AIName` = 'SmartAI', `ScriptName` = '' WHERE `entry` IN (17900,17901);
+
+-- Create SAI
+-- Cast spell on gossip select (1 for each)
+-- Close gossip
```
Comment (tkrokli) [inline sql/updates/world/3.3.5/2017_09_25_99_world.sql]: Perhaps combine these 3 lines into 1 and use ` / ` as a divider between each statement?

Comment (velinath) [reply sql/updates/world/3.3.5/2017_09_25_99_world.sql]: tbh, they're actually artifacted from when I was working on them (as a reminder of what needed to be done), shouldn't even be there at all :P

**Comment by tkrokli** [comment]
OK, so the comment lines are mostly working tools / reminders / todo list. When all is done and checked, I presume they can be removed, save for the top comment line which can be empty ( `-- ` ), but just used as a separator from the previous SQL files in the source when the PR gets merged.

**Comment by velinath** [comment]
Wondering if you have to add cast flag 2 (SMARTCAST_TRIGGERED) to get it to work, although I wouldn't think so.

**Comment by tkrokli** [comment]
Good question, I don't know. I need to think about it and maybe see if I can come up with a suggestion on Wednesday when I got some time off.

**Pull request closed without merging**